tests/test_ote_training.py

```diff
@@ -0,0 +1,613 @@
+import copy
+import glob
+import itertools
+import logging
+import os
+import os.path as osp
+import pytest
+import yaml
+
+from collections import namedtuple
+from copy import deepcopy
+from pprint import pformat
+
+from ote_sdk.configuration.helper import create
+from ote_sdk.entities.inference_parameters import InferenceParameters
+from ote_sdk.entities.metrics import Performance, ScoreMetric
+from ote_sdk.entities.model import (
+    ModelEntity,
+    ModelPrecision,
+    ModelStatus,
+    ModelOptimizationType,
+    OptimizationMethod,
+)
+from ote_sdk.entities.model_template import parse_model_template, TargetDevice
+from ote_sdk.entities.optimization_parameters import OptimizationParameters
+from ote_sdk.entities.resultset import ResultSetEntity
+from ote_sdk.entities.subset import Subset
+from ote_sdk.usecases.tasks.interfaces.export_interface import ExportType
+from ote_sdk.usecases.tasks.interfaces.optimization_interface import OptimizationType
+from ote_sdk.entities.task_environment import TaskEnvironment
+
+from mmdet.apis.ote.apis.detection.config_utils import set_values_as_default
+from mmdet.apis.ote.apis.detection.ote_utils import (generate_label_schema,
+                                                     get_task_class)
+from mmdet.apis.ote.extension.datasets.mmdataset import MMDatasetAdapter
+from sc_sdk.entities.dataset_storage import NullDatasetStorage
+
+from e2e_test_system import e2e_pytest_performance, DataCollector
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+def DATASET_PARAMETERS_FIELDS():
```

Comment:
To make it immutable.
Hmmm.. it returns a list. Ok, I will make it return a tuple, and it will be really immutable!

Reply:
Done. Now it returns tuple!

Reply:
Awesome. Isn't tuple immutable when it's not returned by some function?

Reply:
Because python does not have `const` keyword! if we use just a variable `DATASET_PARAMETERS_FIELDS`, somewhere in the code the value of the variable could be changed. 
Functions returning immutable values is a cheap method of implementing constants in python.